Source/JTSImageViewController.m

```diff
@@ -1692,34 +1700,76 @@ - (void)dismissingPanGestureRecognizerPanned:(UIPanGestureRecognizer *)panner {
         return;
     }
     
+    CGPoint velocity = [panner velocityInView:panner.view];
+    
+    if (JTSImageViewControllerDismissMode_UpDownDirection == self.modalDismissMode) {
+        if (fabs(velocity.y) < fabs(velocity.x)) {
+            // Allow only up/down
+            return;
+        }
+    }
+    
+    //Only used for mode JTSImageViewControllerDismissMode_UpDownDirection
+    CGPoint centerOfView = self.view.center;
+    
     CGPoint translation = [panner translationInView:panner.view];
     CGPoint locationInView = [panner locationInView:panner.view];
-    CGPoint velocity = [panner velocityInView:panner.view];
     CGFloat vectorDistance = sqrtf(powf(velocity.x, 2)+powf(velocity.y, 2));
-    
+
     if (panner.state == UIGestureRecognizerStateBegan) {
+        
         _flags.isDraggingImage = CGRectContainsPoint(self.imageView.frame, locationInView);
+        
         if (_flags.isDraggingImage) {
+            
+            if (JTSImageViewControllerDismissMode_UpDownDirection == self.modalDismissMode) {
+                // Change the center for JTSImageViewControllerDismissMode_UpDownDirection
+                locationInView = centerOfView;
+            }
+            
             [self startImageDragging:locationInView translationOffset:UIOffsetZero];
+
         }
     }
     else if (panner.state == UIGestureRecognizerStateChanged) {
+        
         if (_flags.isDraggingImage) {
+            
             CGPoint newAnchor = self.imageDragStartingPoint;
-            newAnchor.x += translation.x + self.imageDragOffsetFromActualTranslation.horizontal;
-            newAnchor.y += translation.y + self.imageDragOffsetFromActualTranslation.vertical;
+            
+            if (JTSImageViewControllerDismissMode_UpDownDirection == self.modalDismissMode) {
+                newAnchor.x = centerOfView.x;
+                newAnchor.y += translation.y;
+            } else {
+                newAnchor.x += translation.x + self.imageDragOffsetFromActualTranslation.horizontal;
+                newAnchor.y += translation.y + self.imageDragOffsetFromActualTranslation.vertical;
+            }
+            
```

Comment:
I'd like to hear why you went with this overall approach and not something I believe would be simpler and easier to debug. I think you could drop all of these additions and changes in here and just force the x component of `translation` and `velocity` to zero if horizontal movement is disallowed:

```
CGPoint translation = [panner translationInView:panner.view];
CGPoint velocity = [panner velocityInView:panner.view];
if (JTSImageViewControllerDismissMode_UpDownDirection == self.modalDismissMode) {
    translation.x = 0;
    velocity.x = 0;
}
CGPoint locationInView = [panner locationInView:panner.view];
CGFloat vectorDistance = sqrtf(powf(velocity.x, 2)+powf(velocity.y, 2));
```

I think you could then leave the rest of the logic intact. It would be as if the user had entered a perfectly vertical gesture.